### PR TITLE
Mosip 22212 ability to multi select and upload packets in batches

### PR DESCRIPTION
**Pull request by anshulv1401**
Add UI for Multiselect Packet Serialize sync and upload of a packet
Added option to select all
Allow the packet to be uploaded only if n/w is good. Show progress bar for uploading packets Implement data binded list
PacketUploadService with Queue sync and upload operation added
Upload_Over_Wifi_Only configuration added
PacketUploadService Test cases

### CODE REVIEW

File: client/app/src/main/java/io/mosip/registration/app/activites/PacketListActivity.java [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+        if (item.getItemId() == android.R.id.home) {
+            finish();
+        } else {
+            Intent i = new Intent(PacketListActivity.this, SettingsActivity.class);
+            startActivity(i);
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    private boolean checkNetwork() {
```
Comment (ase-101) [inline client/app/src/main/java/io/mosip/registration/app/activites/PacketListActivity.java:288]: This should be part of some Util class as we may need to reuse this method in other activities.

Comment (anshulv1401) [reply client/app/src/main/java/io/mosip/registration/app/activites/PacketListActivity.java:283]: Network Utility methods added in NetworkUtil class.

File: client/app/src/main/java/io/mosip/registration/app/activites/PacketListActivity.java [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+    NetworkUtil networkUtil;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        bi = DataBindingUtil.setContentView(this, R.layout.packet_list_activity);
+        networkUtil = new NetworkUtil(this);
+
+        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this /* Activity context */);
+        uploadOverWifiOnly = sharedPreferences.getBoolean("wifi_only", false);
```
Comment (ase-101) [inline client/app/src/main/java/io/mosip/registration/app/activites/PacketListActivity.java:82]: Is the idea here to take this as a setting and to store it in the sharedPreferences?

File: client/app/src/main/java/io/mosip/registration/app/activites/PacketListActivity.java [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+    }
+
+    private void init() {
+        actionCallback = new ActionCallback();
+        registrationPacketViewModel = new RegistrationPacketViewModel(packetService, dateUtil);
+        list = registrationPacketViewModel.getList();
+        adapter = new RegistrationPacketListAdapter(this, list);
+
+        bi.packetList.setLayoutManager(new LinearLayoutManager(this));
+        bi.packetList.setHasFixedSize(true);
```
Comment (ase-101) [inline client/app/src/main/java/io/mosip/registration/app/activites/PacketListActivity.java:115]: Why is this set to true?

File: client/app/src/main/java/io/mosip/registration/app/activites/PacketListActivity.java [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+        if (count == 0) {
+            actionMode.finish();
+        } else {
+            actionMode.setTitle(String.valueOf(count));
+            actionMode.invalidate();
+        }
+    }
+
+    private void syncAndUploadSelectedPacket() {
+        if (!checkNetwork()) {
```
Comment (ase-101) [inline client/app/src/main/java/io/mosip/registration/app/activites/PacketListActivity.java:161]: Need to notify the user about the unavailability of the network.

Comment (anshulv1401) [reply client/app/src/main/java/io/mosip/registration/app/activites/PacketListActivity.java:161]: Toast msg is raised in "checkNetwork" method. line 291,296

File: client/app/src/main/java/io/mosip/registration/app/activites/PacketListActivity.java (hunk 2 above)
Comment (anshulv1401) [reply client/app/src/main/java/io/mosip/registration/app/activites/PacketListActivity.java:82]: this is a shared preferences setting. loading the value into uploadOverWifiOnly variable on the activity start and resume

File: client/app/src/main/java/io/mosip/registration/app/activites/PacketListActivity.java (hunk 3 above)
Comment (anshulv1401) [reply client/app/src/main/java/io/mosip/registration/app/activites/PacketListActivity.java]: Removed

**Review by ase-101 — APPROVED** [review]

**Pull request merged**